### PR TITLE
Removed floor and ceil implementation from runtime library

### LINKED ISSUE #3292: Remove `floor` & `ceil` implementation from runtime library
With #3270 and #3283 merged we have implementation of `floor` and `ceiling` respectively in `intrinsic_function_registry`.

Hence, we can move ahead and remove `floor` from `lfortran_intrinsic_math3.f90` and `ceiling` from `lfortran_intrinsic_math2.f90`

Let's make two PRs for this:

* [ ] Floor
* [ ] Ceil

> Floor

```diff
diff --git a/src/runtime/CMakeLists.txt b/src/runtime/CMakeLists.txt
index 861d5ef33..3f94497bc 100644
--- a/src/runtime/CMakeLists.txt
+++ b/src/runtime/CMakeLists.txt
@@ -36,13 +36,9 @@ lfortran_compile_runtime(lfortran_intrinsic_custom custom)
 lfortran_compile_runtime(lfortran_intrinsic_optimization builtin)
 add_dependencies(lfortran_intrinsic_optimization lfortran_intrinsic_iso_fortran_env)
 
-lfortran_compile_runtime(lfortran_intrinsic_math3 pure)
-add_dependencies(lfortran_intrinsic_math3 lfortran_intrinsic_iso_fortran_env)
-add_dependencies(lfortran_intrinsic_math3 lfortran_intrinsic_builtin)
-
 lfortran_compile_runtime(lfortran_intrinsic_math2 pure)
 add_dependencies(lfortran_intrinsic_math2 lfortran_intrinsic_iso_fortran_env)
-add_dependencies(lfortran_intrinsic_math2 lfortran_intrinsic_math3)
+# add_dependencies(lfortran_intrinsic_math2)
 add_dependencies(lfortran_intrinsic_math2 lfortran_intrinsic_builtin)
 
 lfortran_compile_runtime(lfortran_intrinsic_ieee_arithmetic pure)
@@ -88,7 +84,6 @@ install(
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_ieee_arithmetic.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_iso_c_binding.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_math2.mod
-  ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_math3.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_math.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_trig.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_sin.mod
diff --git a/src/runtime/pure/CMakeLists.txt b/src/runtime/pure/CMakeLists.txt
index f1866a621..0299dbf11 100644
--- a/src/runtime/pure/CMakeLists.txt
+++ b/src/runtime/pure/CMakeLists.txt
@@ -6,7 +6,6 @@ add_library(lfortran_runtime_pure
     STATIC
     lfortran_intrinsic_iso_c_binding.f90
     lfortran_intrinsic_math2.f90
-    lfortran_intrinsic_math3.f90
     lfortran_intrinsic_iso_fortran_env.f90
     lfortran_intrinsic_ieee_arithmetic.f90
     lfortran_intrinsic_trig.f90
diff --git a/src/runtime/pure/lfortran_intrinsic_math2.f90 b/src/runtime/pure/lfortran_intrinsic_math2.f90
index 3004802ec..dfb486312 100644
--- a/src/runtime/pure/lfortran_intrinsic_math2.f90
+++ b/src/runtime/pure/lfortran_intrinsic_math2.f90
@@ -1,7 +1,6 @@
 ! Temporary module, a subset of lfortran_intrinsic_math that works
 module lfortran_intrinsic_math2
 use, intrinsic :: iso_fortran_env, only: i8 => int8, i16 => int16, i32 => int32, i64 => int64, sp => real32, dp => real64
-use lfortran_intrinsic_math3, only: floor
 implicit none
 
 interface sqrt
diff --git a/src/runtime/pure/lfortran_intrinsic_math3.f90 b/src/runtime/pure/lfortran_intrinsic_math3.f90
deleted file mode 100644
index 0e0034976..000000000
--- a/src/runtime/pure/lfortran_intrinsic_math3.f90
+++ /dev/null
@@ -1,53 +0,0 @@
-module lfortran_intrinsic_math3
-use, intrinsic :: iso_fortran_env, only: i8 => int8, i16 => int16, i32 => int32, i64 => int64, sp => real32, dp => real64
-implicit none
-
-interface floor
-    module procedure sfloor_i32, sfloor_i64, dfloor_i32, dfloor_i64
-end interface
-
-contains
-
-! floor ------------------------------------------------------------------------
-
-elemental integer(i32) function sfloor_i32(x, kind) result(r)
-real(sp), intent(in) :: x
-integer(i32), intent(in) :: kind
-if (x >= 0) then
-    r = x
-else
-    r = x-1
-end if
-end function
-
-elemental integer(i64) function sfloor_i64(x, kind) result(r)
-real(sp), intent(in) :: x
-integer(i64), intent(in) :: kind
-if (x >= 0) then
-    r = x
-else
-    r = x-1
-end if
-end function
-
-elemental integer(i32) function dfloor_i32(x, kind) result(r)
-real(dp), intent(in) :: x
-integer(i32), intent(in) :: kind
-if (x >= 0) then
-    r = x
-else
-    r = x-1
-end if
-end function
-
-elemental integer(i64) function dfloor_i64(x, kind) result(r)
-real(dp), intent(in) :: x
-integer(i64), intent(in) :: kind
-if (x >= 0) then
-    r = x
-else
-    r = x-1
-end if
-end function
-
-end module
diff --git a/src/runtime/tests/CMakeLists.txt b/src/runtime/tests/CMakeLists.txt
index e0adff850..386bc7d02 100644
--- a/src/runtime/tests/CMakeLists.txt
+++ b/src/runtime/tests/CMakeLists.txt
@@ -25,7 +25,6 @@ set(SRC
     ../impure/lfortran_intrinsic_math.f90
     ../impure/lfortran_intrinsic_bit.f90
     ../pure/lfortran_intrinsic_math2.f90
-    ../pure/lfortran_intrinsic_math3.f90
     ../pure/lfortran_intrinsic_string.f90
     ../impure/lfortran_intrinsic_sin.f90
     ../impure/lfortran_intrinsic_sin_c.c
```

> Ceiling

```diff
diff --git a/src/runtime/pure/lfortran_intrinsic_math2.f90 b/src/runtime/pure/lfortran_intrinsic_math2.f90
index dfb486312..679a31308 100644
--- a/src/runtime/pure/lfortran_intrinsic_math2.f90
+++ b/src/runtime/pure/lfortran_intrinsic_math2.f90
@@ -11,10 +11,6 @@ interface aimag
     module procedure saimag, daimag
 end interface
 
-interface ceiling
-    module procedure sceiling, dceiling
-end interface
-
 interface nint
     module procedure snint, dnint
 end interface
@@ -87,24 +83,6 @@ r = 3
 !error stop "aimag not implemented yet"
 end function
 
-! ceiling ------------------------------------------------------------------------
-
-elemental integer function sceiling(x) result(r)
-real(sp), intent(in) :: x
-r = floor(x)
-if (r - x /= 0.0) then
-    r = r + 1
-end if
-end function
-
-elemental integer function dceiling(x) result(r)
-real(dp), intent(in) :: x
-r = floor(x)
-if (r - x /= 0.0) then
-    r = r + 1
-end if
-end function
-
 ! nint ------------------------------------------------------------------------
 
 elemental integer function snint(x) result(r)
```

### PR DESCRIPTION
**Pull request by parth121101**
Fix #3292

### CODE REVIEW

File: src/runtime/CMakeLists.txt [hunk 1]
```diff
@@ -36,13 +36,9 @@ lfortran_compile_runtime(lfortran_intrinsic_custom custom)
...
 add_dependencies(lfortran_intrinsic_optimization lfortran_intrinsic_iso_fortran_env)
 
-lfortran_compile_runtime(lfortran_intrinsic_math3 pure)
-add_dependencies(lfortran_intrinsic_math3 lfortran_intrinsic_iso_fortran_env)
-add_dependencies(lfortran_intrinsic_math3 lfortran_intrinsic_builtin)
-
 lfortran_compile_runtime(lfortran_intrinsic_math2 pure)
 add_dependencies(lfortran_intrinsic_math2 lfortran_intrinsic_iso_fortran_env)
-add_dependencies(lfortran_intrinsic_math2 lfortran_intrinsic_math3)
+# add_dependencies(lfortran_intrinsic_math2)
```
Comment (Thirumalai-Shaktivel) [inline src/runtime/CMakeLists.txt:41]:
```suggestion
```

File: src/runtime/CMakeLists.txt [hunk 2]
```diff
@@ -7,9 +7,9 @@ project(lfortran_unif_runtime C Fortran)
 set(LFORTRAN_NESTED_BUILD no
     CACHE BOOL "Whether we are being called from the LFortran build system")
 
-if(NOT LFORTRAN_NESTED_BUILD)
-  add_subdirectory(legacy)
-endif()
+# if(NOT LFORTRAN_NESTED_BUILD)lfortran_compile_runtime(lfortran_intrinsic_math3 pure)
+#   add_subdirectory(legacy)
+# endif()
```
Comment (Thirumalai-Shaktivel) [inline src/runtime/CMakeLists.txt:12]:
```suggestion
if(NOT LFORTRAN_NESTED_BUILD)
  add_subdirectory(legacy)
endif()
```

**Review by Thirumalai-Shaktivel — COMMENTED** [review]
Also, remove from `comptime_eval.h`.

**Comment by certik** [comment]
Click on "Ready for review" once it is ready.

**Review by certik — APPROVED** [review]
I think that's good, thanks!

**Pull request merged**